File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsCourseAccessibility.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

@Serializable
data class CourseAccessibilityInfo(
    val altTextCoverage: AltTextCoverage = AltTextCoverage(),
    val hardBlockSave: Boolean = false,
)

@Serializable
data class AltTextCoverage(
    val withAlt: Int = 0,
    val total: Int = 0,
    val percent: Int = 100,
    val uncoveredItems: List<UncoveredAccessibilityItem> = emptyList(),
)

@Serializable
data class UncoveredAccessibilityItem(
    val itemId: String,
    val title: String,
    val kind: String,
    val withAlt: Int = 0,
    val total: Int = 0,
    val missing: Int = 0,
)

@Serializable
data class AltTextSuggestion(
    val suggestion: String = "",
    val confidence: Double = 0.0,
)

@Serializable
data class PatchItemMarkdownBody(
    val markdown: String,
)

@Serializable
data class SuggestAltTextBody(
    val imageUrl: String,
    val language: String = "",
)

File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsCourseAccessibility.swift
import Foundation

/// Course accessibility / alt-text models (M13.8).
struct CourseAccessibilityInfo: Codable, Hashable {
    var altTextCoverage: AltTextCoverage
    var hardBlockSave: Bool
}

struct AltTextCoverage: Codable, Hashable {
    var withAlt: Int
    var total: Int
    var percent: Int
    var uncoveredItems: [UncoveredAccessibilityItem]
}

struct UncoveredAccessibilityItem: Codable, Identifiable, Hashable {
    var itemId: String
    var title: String
    var kind: String
    var withAlt: Int
    var total: Int
    var missing: Int

    var id: String { itemId }
}

struct AltTextSuggestion: Codable {
    var suggestion: String
    var confidence: Double
}

struct PatchItemMarkdownBody: Encodable {
    var markdown: String
}

struct SuggestAltTextBody: Encodable {
    var imageUrl: String
    var language: String
}
